packages/cma-client-browser/src/utils/uploadFileOrBlobAndReturnPath.ts

```diff
@@ -34,12 +34,18 @@ export function uploadFileOrBlobAndReturnPath(
   fileOrBlob: File | Blob,
   options: Options = {},
 ): CancelablePromise<string> {
-  if (!(options.filename && 'name' in fileOrBlob)) {
+  let filename: string;
+
+  if (options.filename) {
+    // If a filename is explicitly provided, use it
+    filename = options.filename;
+  } else if (fileOrBlob instanceof File && fileOrBlob.name) {
+    // Files extend Blobs and may have a `name` property. Blobs never have names.
+    filename = fileOrBlob.name;
+  } else {
```

Comment:
We could've also just changed the `&&` to `||` in the original `if`, but then that will cause the later line `filename = options.filename || fileOrBlob.name` to have a TypeScript error, because blobs can never have names.